--- makefile~	2007-04-16 16:45:21.000000000 +0200
+++ makefile	2009-04-20 17:23:54.000000000 +0200
@@ -19,7 +19,7 @@
 # -O2, -O3 ... to find the best optimization level.
 # ===================================================================
 
-CXXFLAGS = -g
+CXXFLAGS = -O2
 
 # PREDCXXFLAGS is for compiling J. Shewchuk's predicates. It should
 # always be equal to -O0 (no optimization). Otherwise, TetGen may not
